### DISCUSSION TITLE
php 8.4: fails with Implicitly marking parameter $argv as nullable is deprecated, the explicit nullable type must be used instead

### DISCUSSION THREAD

**Discussion by toxsick** [deployphp/deployer#3963] [Bugs] [answered]
### Deployer Version

v7.5.5

### Target OS

Alpine v3.20

### Which PHP version are you using?

PHP 8.4

### Content of deploy.php or deploy.yaml

just use latest composer lts docker image, which uses  php 8.4.x, see:

https://dev.to/gromnan/fix-php-84-deprecation-implicitly-marking-parameter-as-nullable-is-deprecated-the-explicit-nullable-type-must-be-used-instead-5gp3

### Steps to reproduce

$ /tmp/vendor/bin/dep --version
Deprecated: React\Promise\Timer\timeout(): Implicitly marking parameter $loop as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/react/promise-timer/src/functions.php on line 142
Deprecated: React\Promise\Timer\sleep(): Implicitly marking parameter $loop as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/react/promise-timer/src/functions.php on line 229
Deprecated: React\Promise\Timer\resolve(): Implicitly marking parameter $loop as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/react/promise-timer/src/functions.php on line 287
Deprecated: React\Promise\Timer\reject(): Implicitly marking parameter $loop as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/react/promise-timer/src/functions.php on line 330
Fatal error: Uncaught ErrorException: Symfony\Component\Console\Input\ArgvInput::__construct(): Implicitly marking parameter $argv as nullable is deprecated, the explicit nullable type must be used instead in phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/symfony/console/Input/ArgvInput.php:46
Stack trace:
#0 phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/composer/ClassLoader.php(444): {closure:phar:///tmp/vendor/deployer/deployer/bin/dep/bin/dep:79}(8192, 'Symfony\\Compone...', 'phar:///tmp/ven...', 46)
#1 phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/composer/ClassLoader.php(444): include()
#2 phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('phar:///tmp/ven...')
#3 phar:///tmp/vendor/deployer/deployer/bin/dep/src/Deployer.php(304): Composer\Autoload\ClassLoader->loadClass('Symfony\\Compone...')
#4 phar:///tmp/vendor/deployer/deployer/bin/dep/bin/dep(96): Deployer\Deployer::run('7.5.5', '')
#5 /tmp/vendor/deployer/deployer/bin/dep(4): require('phar:///tmp/ven...')
#6 /tmp/vendor/bin/dep(119): include('/tmp/vendor/dep...')
#7 {main}
  thrown in phar:///tmp/vendor/deployer/deployer/bin/dep/vendor/symfony/console/Input/ArgvInput.php on line 46

**Comment by garak** [comment]
see https://github.com/orgs/reactphp/discussions/589

**Comment by gremo** [comment]
Got the same error. How to update and fix it?

**Reply by antonmedv** [reply]
I need to implement the fix for the v7 version. I need to find time to do it.

**Reply by gremo** [reply]
Can we help somehow?

**Reply by garak** [reply]
I don't want to underestimate the problem, but... what about updating the erroring vendors to their latest versions?

**Reply by gremo** [reply]
@garak I think it's not that easy. There are workflows and docker images to update too (the PHP version need to be changed or added there).

**Reply by garak** [reply]
How hard could it be? I updated some php docker images myself, and it was just changing "8.3" to "8.4" and checking a couple of incompatible extensions (most notably xdebug).

**Reply by gremo** [reply]
Well how knows? Let's wait for @antonmedv 

I'm thrilled to start using PHP 8.4 as well, but for now, Deployer is the only library holding me back from doing so 😢

**Reply by antonmedv** [reply]
I have to do some fixes for Sympfony  deps we are using (can't simply update deps thought).

Let's buy me some coffee via https://github.com/sponsors/antonmedv, to boost my motivation! ;)

**Comment by gremo** [comment, marked as answer]
Latest version 7.5.7 seems to work fine 🥳
